src/librustc/ty/context.rs

```diff
@@ -2060,4 +2113,16 @@ pub fn provide(providers: &mut ty::maps::Providers) {
         let id = tcx.hir.definitions().def_index_to_hir_id(id.index);
         tcx.stability().local_deprecation_entry(id)
     };
+    providers.extern_mod_stmt_cnum = |tcx, id| {
+        let id = tcx.hir.definitions().find_node_for_hir_id(id);
+        tcx.cstore.extern_mod_stmt_cnum_untracked(id)
+    };
+    providers.all_crate_nums = |tcx, cnum| {
+        assert_eq!(cnum, LOCAL_CRATE);
+        Rc::new(tcx.cstore.crates_untracked())
+    };
+    providers.postorder_cnums = |tcx, cnum| {
```

Comment:
Oh, right, we can do this now! @nikomatsakis do you think we should make move to this style more uniformly?